src/tests/ngraph_helpers/snippets_ngraph_functions/include/precision_propagation_function.hpp

```diff
@@ -0,0 +1,159 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <memory>
+#include <ngraph/ngraph.hpp>
+#include "ngraph/opsets/opset1.hpp"
+#include "snippets/op/convert_saturation.hpp"
+
+namespace ov {
+namespace test {
+namespace snippets {
+
+class PrecisionPropagationFunction {
+public:
+    template<typename T>
+    static std::shared_ptr<ngraph::Function> get(
+        const ngraph::element::Type precision1,
+        const ngraph::PartialShape& inputShape1,
+        const ngraph::element::Type precision2,
+        const ngraph::PartialShape& inputShape2,
+        const ngraph::element::Type constant_precision,
+        const std::pair<element::Type, element::Type>& convertion_before_op1 = std::pair<element::Type, element::Type>(),
+        const element::Type convertion_before_op2_1 = element::undefined,
+        const std::pair<element::Type, element::Type>& convertion_before_op2_2 = std::pair<element::Type, element::Type>(),
+        const element::Type convertion_after_op2 = {}) {
+        const auto branch1 = make_branch(precision1, inputShape1, 1, convertion_before_op1.first);
+        const auto branch2 = make_branch(precision2, inputShape2, 2, convertion_before_op1.second);
+
+        std::shared_ptr<Node> parent = std::make_shared<T>(branch1.second, branch2.second);
```

Comment:
_'If you want to create different nodes in the future, you can just pass them as an argument.'_
Custom operation type is a part of function parameterization approach which is defined outside of function. With template it's more flexible.

_'This way we can avoid duplication of the whole function body. We don't need this code duplication, since the function will be called just a couple times.'_
In this case we mix function parameterization and function implementation, add `if`/`else` in function implementation. Template instantiation is not _'code dublication'_. 

Note, please, `SnippetsFunctionBase` and many of its useless empty inherited types with intermediate members increase binary size more then just static methods.